Make it possible again to see item tooltips on Android

grorp opened the pull request:
This PR is a quick fix so that item tooltips show again on Android. Fixes (<- Github, don't link this please) https://github.com/minetest/minetest/issues/13822#issuecomment-1773789974, a regression from #13146.

Please test this PR and let me know whether you think that the new behavior of the tooltips makes sense.

## To do

This PR is a Ready for Review.

## How to test

Play Minetest on a desktop computer. Verify that item tooltips still work the same as before.

Play Minetest on Android. Verify that it is possible to see item tooltips.

srifqi reviewed (approved):
This PR looks good to me.

https://github.com/minetest/minetest/assets/4017302/90835e51-938a-4497-97eb-4c7753803ed9

-------

<details>
<summary>Probably off-topic</summary>

There are some problems (outside this PR) that makes it somewhat annoying to use, but those are for another PR. For example, needing to tap back to put it into the original/source slot (which happens quite often in Android, especially with this PR, for seeing tooltip). Maybe making the tooltip stays drawn until the selected stack is put back down? (I do not know how to do it.)
</details>

grorp commented:
> Maybe making the tooltip stays drawn until the selected stack is put back down?

That would be possible, but I'm worried that it would make the tooltips (especially those with a lot of text) too annoying. Also, I find it confusing that the highlight is gone, but the tooltip is still there. If we do this, we should make sure that the highlight stays visible as long as the tooltip is still displayed.

```patch
diff --git a/src/gui/guiFormSpecMenu.h b/src/gui/guiFormSpecMenu.h
index 5641e39a9a..8df1caee87 100644
--- a/src/gui/guiFormSpecMenu.h
+++ b/src/gui/guiFormSpecMenu.h
@@ -286,6 +286,8 @@ class GUIFormSpecMenu : public GUIModalMenu
        bool getAndroidUIInput();
 #endif
 
+       v2s32 getPointerPos() { return m_pointer; };
+
 protected:
        v2s32 getBasePos() const
        {
diff --git a/src/gui/guiInventoryList.cpp b/src/gui/guiInventoryList.cpp
index 6f1d28e0b0..86b0f2b798 100644
--- a/src/gui/guiInventoryList.cpp
+++ b/src/gui/guiInventoryList.cpp
@@ -154,7 +154,8 @@ void GUIInventoryList::draw()
                bool show_tooltip = !item.empty() && hovering && !selected_item;
 #ifdef HAVE_TOUCHSCREENGUI
                // Make it possible to see item tooltips on touchscreens
-               show_tooltip |= hovering && selected && m_fs_menu->getSelectedAmount() != 0;
+               show_tooltip |= rect.isPointInside(m_fs_menu->getPointerPos()) &&
+                               selected && m_fs_menu->getSelectedAmount() != 0;
 #endif
                if (show_tooltip) {
                        std::string tooltip = orig_item.getDescription(client->idef());
```

srifqi commented:
> ... but I'm worried that it would make the tooltips (especially those with a lot of text) too annoying.

I agree. That was my random thought on this issue. It is good as it is now for me.

-------

By the way, what does the last commit (db713ab) do?

grorp commented:
> By the way, what does the last commit (db713ab) do?

1. I found an outdated TODO comment related to tooltips on Android and replaced it with a useful comment.

3. I added a check so that no tooltip is drawn if the selected itemstack is empty. I had assumed this was impossible, but it turns out it can happen during "left-dragging to distribute evenly".

    https://github.com/minetest/minetest/blob/db713abea19aff02f3b34d2953ed6f551aee3e42/src/gui/guiFormSpecMenu.cpp#L3775-L3781
    https://github.com/minetest/minetest/blob/db713abea19aff02f3b34d2953ed6f551aee3e42/src/gui/guiFormSpecMenu.cpp#L3854-L3859

SmallJoker commented:
:+1: LGTM
